src/settings/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Copyright © 2021-2022 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+add_library(settings STATIC
+        basic_setting_spec.cpp
+        bool_setting_spec.cpp
+        persistent_settings_handler.cpp
+        settings.cpp
+        dynamic_setting_spec.cpp)
```

Comment:
Weird ordering?

Reply:
Oops, I'm blaming that on CLion ;) Fixed with the renaming to "custom"